types/time.go

```diff
@@ -298,27 +299,6 @@ func (t Time) Compare(o Time) int {
 	return compareTime(t.Time, o.Time)
 }
 
-func compareTime(a, b MysqlTime) int {
```

Comment:
Please move compareTime back here, keep mytime.go clean.
You can add anything in time.go, but it's better don't touch mytime.go.